src/session_info2/_dists.py

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: MPL-2.0
+from __future__ import annotations
+
+import re
+from importlib.metadata import Distribution, distributions
+from importlib.metadata import packages_distributions as pkgs_dists
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Generator, Mapping
+
+
+def packages_distributions() -> Mapping[str, list[str]]:
+    """Return a mapping of top-level packages to their distributions.
+
+    Unlike :func:`importlib.metadata.packages_distributions`,
+    this includes editable packages.
+    """
+    pds = dict(pkgs_dists())
+    for dist in distributions():
+        for pkg_name in _top_level_editable(dist):
+            if "." not in pkg_name:  # apparently that’s what makes an importable name
+                pds.setdefault(pkg_name, []).append(dist.name)
+    return pds
+
+
+def _top_level_editable(dist: Distribution) -> Generator[str, None, None]:
+    """Find top-level packages in an editable distribution."""
+    for pth_file in dist.files or ():
```

Comment:
Not sure I follow what is in `files` given https://docs.python.org/3/library/importlib.metadata.html#importlib.metadata.Distribution.  It's just all the files in the package or?

Reply:
yeah:

<img width="804" height="91" alt="image" src="https://github.com/user-attachments/assets/dc7dfcb3-349c-4d8a-ac5f-26b887dd387d" />

also a candidate for https://github.com/python/cpython/issues/110937